dao/mongodb.rya/src/main/java/mvm/rya/mongodb/instance/MongoDetailsAdapter.java

```diff
@@ -0,0 +1,143 @@
+import com.mongodb.BasicDBList;
+import com.mongodb.BasicDBObject;
+import com.mongodb.BasicDBObjectBuilder;
+import com.mongodb.DBObject;
+
+import mvm.rya.api.instance.RyaDetails;
+import mvm.rya.api.instance.RyaDetails.EntityCentricIndexDetails;
+import mvm.rya.api.instance.RyaDetails.FreeTextIndexDetails;
+import mvm.rya.api.instance.RyaDetails.GeoIndexDetails;
+import mvm.rya.api.instance.RyaDetails.JoinSelectivityDetails;
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails;
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails.FluoDetails;
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails.PCJDetails;
+import mvm.rya.api.instance.RyaDetails.PCJIndexDetails.PCJDetails.PCJUpdateStrategy;
+import mvm.rya.api.instance.RyaDetails.ProspectorDetails;
+import mvm.rya.api.instance.RyaDetails.TemporalIndexDetails;
+
+/**
+ * Serializes configuration details for use in Mongo.
+ * The {@link DBObject} will look like:
+ * <pre>
+ * {@code
+ * {
+ *   "instanceName": &lt;string&gt;,
+ *   "version": &lt;string&gt;?,
+ *   "entityCentricDetails": &lt;boolean&gt;,
+ *   "geoDetails": &lt;boolean&gt;,
+ *   "pcjDetails": {
+ *       "enabled": &lt;boolean&gt;,
+ *       "fluoName": &lt;string&gt;,
+ *       "pcjs": [{
+ *           "id": &lt;string&gt;,
+ *           "updateStrategy": &lt;string&gt;,
+ *           "lastUpdate": &lt;date&gt;
+ *         },...,{}
+ *       ]
+ *   },
+ *   "temporalDetails": &lt;boolean&gt;,
+ *   "freeTextDetails": &lt;boolean&gt;,
+ *   "prospectorDetails": &lt;date&gt;,
+ *   "joinSelectivityDetails": &lt;date&gt;
+ * }
+ * </pre>
+ */
+public class MongoDetailsAdapter {
+    public static final String INSTANCE_KEY = "instanceName";
+    public static final String VERSION_KEY = "version";
+
+    public static final String ENTITY_DETAILS_KEY = "entityCentricDetails";
+    public static final String GEO_DETAILS_KEY = "geoDetails";
+    public static final String PCJ_DETAILS_KEY = "pcjDetails";
+    public static final String PCJ_ENABLED_KEY = "enabled";
+    public static final String PCJ_FLUO_KEY = "fluoName";
+    public static final String PCJ_PCJS_KEY = "pcjs";
+    public static final String PCJ_ID_KEY = "id";
+    public static final String PCJ_UPDATE_STRAT_KEY = "updateStrategy";
+    public static final String PCJ_LAST_UPDATE_KEY = "lastUpdate";
+    public static final String TEMPORAL_DETAILS_KEY = "temporalDetails";
+    public static final String FREETEXT_DETAILS_KEY = "freeTextDetails";
+
+    public static final String PROSPECTOR_DETAILS_KEY = "prospectorDetails";
+    public static final String JOIN_SELECTIVITY_DETAILS_KEY = "joinSelectivitiyDetails";
+
+    /**
+     * Serializes {@link RyaDetails} to mongo {@link DBObject}.
+     * @param details - The details to be serialized.
+     * @return The mongo {@link DBObject}.
+     */
+    public static BasicDBObject serialize(final RyaDetails details) {
+        final PCJIndexDetails pcjDetails = details.getPCJIndexDetails();
```

Comment:
Maybe move all the PCJIndexDetails object conversion to its own method. That would make this a lot easier to read.